src/gsy_e/gsy_e_core/simulation/results_manager.py

```diff
@@ -157,20 +157,20 @@ def update_scm_manager(self, scm_manager: SCMManager) -> None:
         self._scm_manager = scm_manager
 
     @classmethod
-    def _update_area_stats(cls, area: "Area", endpoint_buffer: "SimulationEndpointBuffer") -> None:
+    def _update_area_stats(
+            cls, area: "CoefficientArea", endpoint_buffer: "SimulationEndpointBuffer") -> None:
         return
 
-    def update_csv_files(self, slot_no: int, current_time_slot: DateTime, area: "Area",
+    def update_csv_files(self, slot_no: int, current_time_slot: DateTime, area: "CoefficientArea",
                          scm_manager: SCMManager) -> None:
         """Update the csv results on market cycle."""
         if self.export_results_on_finish:
             self._export.data_to_csv(area, current_time_slot, slot_no == 0, scm_manager)
 
-    def save_csv_results(self, area: "Area") -> None:
+    def save_csv_results(self, area: "CoefficientArea") -> None:
         """Update the CSV results on finish, and write the CSV files."""
         if self.export_results_on_finish:
             log.info("Exporting simulation data.")
-            self._export.data_to_csv(area, False, None)
```

Comment:
How come? Was it due to an error on the generation of the CSV files ?

Reply:
yes, the last market slot was written twice to the CSV files.